docs/src/caliptra_util_host_library.md

```diff
@@ -0,0 +1,540 @@
+
+// Sign hash with ECDSA
+caliptra_error_t caliptra_cmd_ecdsa_sign(caliptra_session_t* session, const uint8_t* hash, size_t hash_len, caliptra_ecdsa_signature_t* signature, caliptra_ecdsa_pubkey_t* pubkey);
+```
+
+#### Telemetry Commands (WIP)
+
+```c
+// Get log data
+caliptra_error_t caliptra_cmd_get_log(caliptra_session_t* session, caliptra_log_type_t log_type, uint8_t* buffer, size_t* buffer_size);
+
+// Clear log data
+caliptra_error_t caliptra_cmd_clear_log(caliptra_session_t* session, caliptra_log_type_t log_type);
+```
+
+## Extensibility with Plugin Architecture
+
+The Caliptra Utility host library is designed to be highly extensible, supporting the dynamic addition of new commands, protocols, and transport mechanisms at runtime through a **plugin architecture**. This approach enables third-party developers and system integrators to extend the library’s capabilities without modifying or recompiling the core library.
+
+```mermaid
+flowchart TD
+    subgraph App["Utility Application"]
+        A1[Main App/CLI]
+        A2["Plugin (Shared Library)"]
+    end
+
+    subgraph Library["Caliptra Utility Host Library"]
+        B1[Command Layer]
+        B2[Core Layer]
+        B3[Transport Abstraction Layer]
+        B4[Transport Impl Layer]
+        B5[Plugin Loader & Registry]
+    end
+
+    A1 -->|Loads at runtime| A2
+    A2 -- Registers new commands/transports/protocols --> B5
+    A1 -->|Uses APIs| B1
+    B1 --> B2
+    B2 --> B3
+    B3 --> B4
+    B5 -- Makes available to --> B1
+```
+
+- **Plugins** are shared libraries (e.g., `.so` or `.dll` files) that implement new commands, protocol handlers, or transport mechanisms.
+- The core library provides APIs to **dynamically load, register, and discover** these plugins at runtime.
+- This design preserves the existing layered architecture, while enabling runtime extensibility and modularity.
+
+### How Plugins Work
+
+1. **Plugin Interface**
+   Each plugin implements a standard interface and exports a descriptor structure. For example:
+   ```c
+   typedef struct {
+       const char* name;
+       const char* version;
+       caliptra_error_t (*register_plugin)(void);
+   } caliptra_plugin_descriptor_t;
+   ```
+   The plugin’s `register_plugin` function is called by the core library to register new commands, transports, or protocol handlers.
+
+2. **Dynamic Loading**
+   The core library provides APIs such as:
+
+   ```c
+   caliptra_error_t caliptra_plugin_load(const char* path);
+   caliptra_error_t caliptra_plugin_unload(const char* name);
+   ```
+   These use platform mechanisms (`dlopen`/`dlsym` on Linux, `LoadLibrary` on Windows) to load plugins at runtime.
+
+3. **Dynamic Registration**
+   Plugins use registration APIs to add new features:
+
+   ```c
+   caliptra_error_t caliptra_command_register(const caliptra_command_desc_t* desc);
```

Comment:
Do we need to have `unregister` functions as well for when the plugins are unloaded?